pkg/controllers/state/statenode.go

```diff
@@ -150,6 +158,54 @@ func (in *StateNode) Pods(ctx context.Context, c client.Client) ([]*v1.Pod, erro
 	return nodeutils.GetPods(ctx, c, in.Node)
 }
 
+// IsDisruptable returns an error if the StateNode cannot be disrupted
+// This checks all associated StateNode internals, node labels, and do-not-disrupt annotations
+// on both the pod and the node.
+// IsDisruptable takes in a recorder to emit events on the nodeclaims when the state node is not a candidate
+//
+//nolint:gocyclo
+func (in *StateNode) IsDisruptable(ctx context.Context, kubeClient client.Client) error {
```

Comment:
nit: I'd expect a bool from this func name

Reply:
We've used the convention of an error to be a bool + string throughout the repo. What would the name need to be for you to think that error should be the return ?

Reply:
(string, bool) is a pretty common go idiom for this, though both `*string` and `error` represent almost the same thing

Reply:
```suggestion
func (in *StateNode) ValidateDisruptable(ctx context.Context, kubeClient client.Client) error {
```

Validate gives more indication that you are returning an error rather than an `Is` statement. We definitely do this in calls like in our disruption validation, so that might be more appropriate here if we are looking for a better function name